### ISSUE TITLE
networkd network fails Could not set NDisc address: Invalid argument

### ISSUE THREAD

**Issue by yu-re-ka** [systemd/systemd#31263]
### systemd version the issue has been seen with

255.2

### Used distribution

NixOS 24.05-pre-git

### Linux kernel version used

6.7.4

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

Expectation: The host is obtaining a DHCPv4 lease and DHCPv6 prefix delegation, and is reachable over the interface with IPv4 and IPv6.

### Unexpected behaviour you saw

The interface enters a failed state (see logs). There is no IPv4 default route. After 30 minutes the host loses its v6 prefix as well and is no longer reachable via IPv4 or IPv6.

### Steps to reproduce the problem

The interface has the following configuration:

```
[Match]
Name=enp4s0

[Network]
DHCP=yes
IPv6PrivacyExtensions=kernel

[DHCPv6]
PrefixDelegationHint=::/60
WithoutRA=solicit
```

### Additional program output

debug log: https://gist.github.com/yu-re-ka/2a2d7c13f17165243dbb63f911d9e4f2

This was previously working with systemd 254
I already tried cherry-picking https://github.com/systemd/systemd/pull/31171/commits/6244184e3ff2f7659406c2d08a9a4d75e08bccb2, but apparently that's not it

**Comment by yu-re-ka** [comment]
CC @yuwata for recent changes to the IPv6 RA / DHCPv6 and NDisc handling code

**Comment by yuwata** [comment]
This is caused by that both NDisc and DHCPv6 try to configure the same address with different prefix length, if the masked IPv6 addresses in the logs are equivalent.

**Comment by yuwata** [comment]
Please set `UseAddress=no` in [DHCPv6] section.

**Comment by yu-re-ka** [comment]
> Please set UseAddress=no in [DHCPv6] section.

That works

**Comment by yu-re-ka** [comment]
It seems you already found the code path that causes this? Let me know if you still need any more information.

**Comment by ssahani** [comment]
> This is caused by that both NDisc and DHCPv6 try to configure the same address with different prefix length, if the masked IPv6 addresses in the logs are equivalent.

@yuwata is correct. Here we are getting  "Invalid argument" from kernel. Can you manually add that address via ``ip address add``` and verify it that works for you. 
```
Feb 09 06:43:54 apu systemd-networkd[1355]: enp4s0: ndisc_check_ready(): SLAAC addresses and routes are not set.
Feb 09 06:43:54 apu systemd-networkd[1355]: enp4s0: Could not set NDisc address: Invalid argument
```

**Comment by yuwata** [comment]
> It seems you already found the code path that causes this? Let me know if you still need any more information.

The issue is a regression caused by 0a0c2672dbd22dc85d660e5baa7e1bef701beb88.

Fix is waiting in #31314.
